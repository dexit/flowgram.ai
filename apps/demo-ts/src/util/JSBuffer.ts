export class JSBuffer {
  buffer: string[] = []

  constructor(public indent = '') {}

  clear(): JSBuffer {
    this.buffer = []
    return this
  }

  push(...args: string[]): JSBuffer {
    this.buffer.push(...args)
    return this
  }

  pushLine(...args: string[]): JSBuffer {
    this.buffer.push(...args, '\n')
    return this
  }

  toJs(separator: string = ''): string {
    // TODO js string perf
    // FIXME last line indent
    return (
      this.buffer
        .join(separator)
        .split('\n')
        .map((line) => `${this.indent}${line}`)
        // .map((line) => (line !== '\n' ? `${this.indent}${line}` : line))
        .join('\n')
    )
  }

  toString(): string {
    return this.toJs()
  }
}
