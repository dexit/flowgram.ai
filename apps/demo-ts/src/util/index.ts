export * from './log'
export * from './JSBuffer'
