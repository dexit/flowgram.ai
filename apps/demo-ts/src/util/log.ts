import debug from 'debug'

export const info = debug('log:info')
export const log = info
export const warn = debug('log:warn')
export const error = debug('log:error')
