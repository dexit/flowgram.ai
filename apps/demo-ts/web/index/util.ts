
export const codeMirrorBaseConfig: CodeMirror.EditorConfiguration = Object.freeze({
  mode: { name: 'javascript', json: true },
  lineNumbers: true,
  lineWrapping: false,
  theme: 'twilight',
  foldGutter: true,
  gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
  readOnly: true,
  cursorBlinkRate: -1,
})