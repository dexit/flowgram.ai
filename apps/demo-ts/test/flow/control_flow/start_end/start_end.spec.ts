import { runCases } from '../../util'

runCases(['min', 'min11', 'min22'], __dirname)
