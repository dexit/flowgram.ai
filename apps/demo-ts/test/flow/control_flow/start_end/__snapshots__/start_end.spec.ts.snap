// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flow min | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {}
  Object.assign(start.outputs, __args)
  //=== end@end | {level:1,blockIndex:1}
  const end = {}
  end.data = {}
  return end.data
}
"
`;

exports[`flow min | {} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
}
`;

exports[`flow min11 | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    query: 'Hello BlockWise',
  }
  Object.assign(start.outputs, __args)
  //=== end@end | {level:1,blockIndex:1}
  const end = {}
  end.data = {
    data: start.outputs.query,
  }
  return end.data
}
"
`;

exports[`flow min11 | {} 2`] = `
{
  "data": "Hello BlockWise",
  "error": {
    "code": "",
    "msg": "",
  },
}
`;

exports[`flow min22 | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    query: 'Hello BlockWise',
    limit: 5,
  }
  Object.assign(start.outputs, __args)
  //=== end@end | {level:1,blockIndex:1}
  const end = {}
  end.data = {
    data: start.outputs.query,
    limit: start.outputs.limit,
  }
  return end.data
}
"
`;

exports[`flow min22 | {} 2`] = `
{
  "data": "Hello BlockWise",
  "error": {
    "code": "",
    "msg": "",
  },
  "limit": 5,
}
`;
